### PR TITLE
even more compact form editor

### PR DESCRIPTION
**Pull request by opqdonut**
follow-up to #1804 #2598

related to #1899

# Checklist for author

Remove items that aren't applicable, check items that are done.

## Reviewability
- [x] Link to issue
- [ ] Consider adding screenshots for ease of review

## Documentation
- [x] Update changelog if necessary
- [ ] Update manual/ (if applicable)

## Testing
- [ ] Valuable features are integration / browser / acceptance tested automatically

### CODE REVIEW

**Comment by opqdonut** [comment]
## Form editor for table field before changes

table-before

## ... after changes

table-after

## ... with additional settings shown

table-after-additional

## ... field collapsed

table-after-collapsed

**Comment by opqdonut** [comment]
Plan:
- [ ] make the field up/down/remove/collapse buttons bigger
- [ ] swap places of the collapse and remove buttons
- [ ] add a confirmation to the remove button
- [ ] make the collapse button turn into an expand button when field is collapsed
- [ ] update manual

**Comment by marharyta** [comment]
I have one comment: It might help to create a small margin to the left side of the additional settings (of the additional setting as they "uncollapse") so that it is easier to see visual hierarchy

**Comment by marharyta** [comment]
See field type

**Comment by opqdonut** [comment]
New screenshots

table-after

table-after-additional

table-after-collapsed

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ Changes since v2.16
   - The external title is shown to applicants in the application and internal name used throughout the administration.
   - The API supports the old style for now.
 - Answers to conditional fields that are not visible are no longer stored by REMS. The API accepts answers for invisible fields but drops them. The UI does not send answers to invisible fields. (#2574)
+- The form editor UI was reworked to look less cluttered. Many inputs are now hidden behind "Show" buttons by default.
```
Comment (Macroz) [inline CHANGELOG.md:23]: Should this link to the issue we have? #1899

File: src/cljs/rems/collapsible.cljs [hunk 2]
```diff
@@ -2,56 +2,82 @@
...
 
+(defn- hide-callback [id callback]
+  (fn [event]
+    (.preventDefault event)
+    (let [element (js/$ (str "#" id))]
+      (.collapse element "show")
+      (.focus element))
+    ;; bootstrap's .collapse returns immediately, so in order to avoid
+    ;; momentarily showing both buttons we wait for a hidden.bs.collapse event
+    (.. (js/$ (str "." id "-more"))
```
Comment (Macroz) [inline src/cljs/rems/collapsible.cljs:13]: I think this part looks nasty but I guess we have to live with it.

Looking forward to the demo!

**Review by Macroz — APPROVED** [review]
I'm happy to see this in action in the next demo!

File: CHANGELOG.md (hunk 1 above)
Comment (opqdonut) [reply CHANGELOG.md]: I guess I didn't add it originally because this doesn't resolve the ticket, but I added the link now. It makes sense.

**Pull request merged**